File: 1695-maximum-erasure-value/1695-maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        int l = 0, res = 0, curr = 0;
        for(int i = 0; i < nums.length; i++) {
            while(!seen.add(nums[i])) {
                curr -= nums[l];
                seen.remove(nums[l]);
                l++;
            }
            curr += nums[i];
            seen.add(nums[i]);
            res = Math.max(res, curr);
        }
        return res;
    }
}

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        #slidng window
        res,curr,l =0, 0, 0
        seen = set()
        for i in range(len(nums)):
            while l <= i and nums[i] in seen:
                curr -= nums[l]
                seen.remove(nums[l])
                l += 1
            seen.add(nums[i])
            curr += nums[i]
            res = max(res, curr)
        return res
            
            
            